src/test/scala/mesosphere/marathon/MarathonSchedulerActorTest.scala

```diff
@@ -108,6 +108,90 @@ class MarathonSchedulerActorTest extends MarathonActorSupport
       awaitAssert({
         driver.reconcileTasks(expectedStatus)
       }, 5.seconds, 10.millis)
+      awaitAssert({
+        driver.reconcileTasks(java.util.Arrays.asList())
+      }, 5.seconds, 10.millis)
+    } finally {
+      stopActor(schedulerActor)
+    }
+  }
+
+  test("Terminal tasks should not be submitted in reconciliation - Instance with only terminal tasks") {
+    val f = new Fixture
+    import f._
+    val app = AppDefinition(id = "/test-app".toPath, instances = 1)
+    val instance = TestInstanceBuilder.newBuilder(app.id)
+      .addTaskError(containerName = Some("error"))
+      .addTaskFailed(containerName = Some("failed"))
+      .addTaskFinished(containerName = Some("finished"))
+      .addTaskKilled(containerName = Some("killed"))
+      .addTaskGone(containerName = Some("gone"))
+      .addTaskDropped(containerName = Some("dropped"))
+      .addTaskUnknown(containerName = Some("unknown"))
+      .getInstance()
+
+    appRepo.ids() returns Source.single(app.id)
+    instanceTracker.instancesBySpec()(any[ExecutionContext]) returns Future.successful(InstanceTracker.InstancesBySpec.of(InstanceTracker.SpecInstances.forInstances("nope".toPath, Iterable(instance))))
```

Comment:
Instead of `"nope".toPath`, shouldn't this be `app.id`? Doesn't matter for the test, but it looks like a wrong setup.